README.md

```diff
@@ -82,141 +143,167 @@ assign res = '{}' | parse_json | hash_merge: redirect_to: redirect_to
-- hook_module_info
-- hook_admin_page
-- hook_permission
+### RBAC Authorization
 
-### Own hooks
+The module provides the fundation for implemeneintg Role-Based Access Control in your application. As described in the registration section, all the user initially receive a `member` role. The first user that signs up will receive a `superadmin` role.
 
-#### hook_user_create
+#### authorization commands 
 
-Fires when the user is created. The created user is added to `params.created_user` and the other sent params are added to `params.hook_params`.
+There are the following helper commands provided by the module to authorize the user:
 
-For example:
+##### can_do command 
+
+It returns either `true` or `false` depending if the user has permission to the operation defined as a `do` argument. Useful to modify the UI based on the permission and do not include functionalities to which user does not have permission
 
 ```
-{% parse_json args %}
-  {
-    "user_id": {{ params.created_user.id | json }},
-    "first_name": {{ params.hook_params.first_name | json }},
-    "last_name": {{ params.hook_params.last_name | json }},
-    "dog_name": {{ params.hook_params.dog_name | json }},
-    "favorite_color": {{ params.hook_params.favorite_color | json }}
-  }
-{% endparse_json %}
-{% liquid
-  function profile = 'modules/user/commands/profiles/create', args: args
-  return profile
-%}
+function can = 'modules/user/helpers/can_do', requester: user, do: 'admin_pages.view'
 ```
 
-#### user_create_validate
+##### can_do_or_unauthorized command 
 
-Fires before the user is created. The command does not attempt to create the user when the input is invalid.
+If the user does not have permission, the 403 page is rendered and the flow stops. Has to use deprecated `include` functionality for the `break` liquid tag to work properly - we do not want to execute code past this point if the user has no permission
 
-The `params` variable holds the input required by the create command.
-It must return an object indicating error if there is any:
-```json
-{
-  "errors": {
-    "fist_name": [
-      "Fist name is too long"
-    ],
-    "favorite_color": [
-      "Given value is not a color: foobar"
-    ]
-  },
-  "valid": false
-}
-```
 
-For example:
 ```
-{% liquid
-  # modules/community/public/lib/hooks/hook_user_create_validate.liquid
-  function object = 'modules/profile/commands/profiles/create/build', object: params
-  function object = 'modules/profile/commands/profiles/create/check', object: object
-
-  return object
-%}
+# platformos-check-disable ConvertIncludeToRender
+include 'modules/user/helpers/can_do_or_unauthorized', requester: current_user, do: 'users.register'
+# platformos-check-enable ConvertIncludeToRender
 ```
 
-#### hook_user_load
+The `platformos-check-disable` and `platformos-check-enable` will let [platformos-check](https://github.com/Platform-OS/platformos-check) to not report a warning for using the `include` tag instead of the `render` tag.
```

Comment:
```suggestion
The `platformos-check-disable` and `platformos-check-enable` tags are used to prevent the [platformOS-check](https://github.com/Platform-OS/platformos-check) from reporting a warning for using the `include` tag instead of the recommended `render` tag.
```